packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -35,6 +40,67 @@ const Indicator = () => {
   const trackedInitialLoad = useRef(false)
   const { track } = useTrackEvent()
 
+  async function fetchPageData() {
+    const urlHostString = window.location.origin
+    const pathAdjustment =
+      window.location.pathname === `/` ? `/index` : window.location.pathname
+
+    const url = `${urlHostString}/page-data${pathAdjustment}/page-data.json`
+    try {
+      const resp = await fetch(url)
+      const data = await resp.text()
+
+      if (data === `{}`) {
+        // for local development, force and error if page is missing.
+        const err = new Error(`Not Found`)
+        err.status = 404
+        throw err
+      }
+
+      return { data, errorMessage: null }
+    } catch (e) {
+      if (e.status === 404) {
+        return { data: null, errorMessage: `This page has moved.` }
+      } else {
+        return { data: null, errorMessage: null }
+      }
+    }
+  }
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const { data } = await fetchPageData()
+      pageData = data
+    }
+
+    fetchData()
+  }, [])
+
+  const hasPageDataChanged = async () => {
+    if (buildId !== latestCheckedBuild) {
+      let pageDataCounter = 0
+      let hasPageChanged = false
+
+      while (!hasPageChanged && pageDataCounter <= PAGE_DATA_RETRY_LIMIT) {
+        const loadedPageData = pageData
+        const { data, errorMessage } = await fetchPageData()
+
+        if (errorMessage) {
+          return { hasPageChanged: false, errorMessage }
+        }
+
+        pageDataCounter++
+        hasPageChanged = loadedPageData !== data
+
+        await new Promise(resolve => setTimeout(resolve, POLLING_INTERVAL))
```

Comment:
```suggestion
        hasPageChanged = loadedPageData !== data

        if (!hasPageChanged) {
          await new Promise(resolve => setTimeout(resolve, POLLING_INTERVAL))
        }
```

We don't need to wait for the `POLLING_INTERVAL` if the page has changed, otherwise it'll just introduce a delay to when they see their updated content

Reply:
Alternatively if you don't want to introduce that `if` statement, you could just move the awaiting poll interval to the start of the `while` loop.